main.cpp

```diff
@@ -56,46 +56,66 @@ Purpose:  This project will show you the difference between member functions and
 #include <string>
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
-    //2
-    //3
+    T(int v, const char* n) : value(v), name(n) {}   //1
+    int value;  //2
+    std::string name;   //3
 };
 
-struct <#structName1#>                                //4
+struct TFunction                               //4
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    T* compare(T* a, T* b) //5
     {
-        if( a->value < b->value ) return a;
-        if( a->value > b->value ) return b;
+        if (a != nullptr && b != nullptr)
+        {
+            if( a->value < b->value ) return a;
+            if( a->value > b->value ) return b;
+        }
         return nullptr;
     }
 };
 
 struct U
 {
-    float <#name1#> { 0 }, <#name2#> { 0 };
-    <#returnType#> <#memberFunction#>(<#type name#>* <#updatedValue#>)      //12
+    float var1 { 0.f }, var2 { 0.f };
+
+    float smoothAndMultiply(float* updatedValue)      //12
     {
-        
+        std::cout << "U's var1 value: " << var1 << std::endl;
+        if (updatedValue != nullptr)
+        {
+            var1 = *updatedValue;
+        }
+        std::cout << "U's var1 updated value: " << var1 << std::endl;
+        while( std::abs(var2 - var1) > 0.001f )
+        {
+            var2 += 0.1f;
+        }
+        std::cout << "U's var2 updated value: " << var2 << std::endl;
+        return var2 * var1;
```

Comment:
This doesn't do the same thing that the static function does. 
your `if()` logic is different.
- [ ] revise